sys/riscv/riscv/pmap.c

```diff
@@ -3435,6 +3435,12 @@ pmap_enter_quick_locked(pmap_t pmap, vm_offset_t va, vm_page_t m,
  * shared mappings) as no other cores are going to have this pmap active.
  *
  * XREF pmap_remove_pages
+ *
+ * XXXMJ the algorithm used here only visits "managed" mappings, i.e., those for
+ * which we maintain reverse mapping info by linking a PV entry with the
+ * physical page.  While most unmanaged mappings refer to device memory, this is
+ * not required.  In particular, OBJT_PHYS objects are unmanaged and back
+ * largepage POSIX shared memory objects (which may contain capabilities).
```

Comment:
To add for confirmation: we think that it's fine for the store-side story to defer synchronization of the MD capdirty bits back to the MI layer until it has discovered an unmanaged mapping during the `vm_cheri_revoke` loop, before it has inspected any pages in that region.